### PR TITLE
Skip probe previous endpoints when HTTPProxy is not valid

### PR DESCRIPTION
**Pull request by nak3**
If Revision was cleaned up by GC before probing the endpoint, Ksvc
never becomes Ready and prober keeps trying to probe forever.

Please see https://github.com/knative-sandbox/net-contour/issues/275 for the detail.

This patch fixes it by skipping probe if HTTPProxy is not valid.

Fixes https://github.com/knative-sandbox/net-contour/issues/275 https://github.com/knative/serving/issues/9582

/cc @mattmoor @whaught

### CODE REVIEW

**Comment by mattmoor** [comment]
Given that this is the first we'll be relying on Contour's status, I'm uncomfortable landing this the day before the release 😅 

Let's hold this and land it post-cut Tuesday, so we have a full cycle to bake it.

/hold

File: pkg/reconciler/contour/resources/kingress.go [hunk 1]
```diff
@@ -51,6 +51,12 @@ func MakeEndpointProbeIngress(ctx context.Context, ing *v1alpha1.Ingress, previo
 
 	// Reverse engineer our previous state from the prior generation's HTTP Proxy resources.
 	for _, proxy := range previousState {
+		// Skip probe when status is not valid. It happens when the previous revision was garbage collected.
+		// see: https://github.com/knative/serving/issues/9582
+		if proxy.Status.CurrentStatus != "valid" {
+			continue
+		}
+
```
Comment (mattmoor) [inline pkg/reconciler/contour/resources/kingress.go:59]: I don't think we log the proxy status anywhere (and really don't key off of this anywhere else).  I'd definitely log here (at INFO), but it'd be useful to get logging around the proxy status in generate (at DEBUG).  That would probably help track this and other issues down via logstream.

/unhold

Comment (nak3) [reply pkg/reconciler/contour/resources/kingress.go:61]: Thank you. I added Info log. But sorry I didn't understand about where is the best place for `it'd be useful to get logging around the proxy status in generate (at DEBUG)`.

Comment (mattmoor) [reply pkg/reconciler/contour/resources/kingress.go:61]: Sorry, I meant that we should put that Info log statement alongside the `continue`.

For the Debug log statement, maybe we should put it in the create/update loop (after it fetches a proxy)?

Comment (nak3) [reply pkg/reconciler/contour/resources/kingress.go]: Alright. I updated.
The create loop already had the debug log https://github.com/knative-sandbox/net-contour/blob/600a323c5b92d260dcd637058bea1fd3c7f9e15a/pkg/reconciler/contour/contour.go#L211 so just added to update loop.

File: pkg/reconciler/contour/contour.go [hunk 2]
```diff
@@ -223,10 +223,11 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, ing *v1alpha1.Ingress) r
 			return err
 		}
 		if diff, err := kmp.SafeDiff(update, matches[0]); err == nil {
-			logger.Debugf("Updated http proxy: %s", diff)
+			logger.Debug("Updated http proxy diff: ", diff)
```
Comment (nak3) [inline pkg/reconciler/contour/contour.go:226]: FYI 
L#226 and L#230 prints like logs this:
https://gist.githubusercontent.com/nak3/ebee40ab5c35a25abf4ea87982c0c3bd/raw/642b22546d5227ee2f527542cedd315d0f365d87/contour-httpproxy-example.log

**Review by mattmoor — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**